alphadia/workflow/searchoptimization.py

```diff
@@ -58,8 +49,140 @@ def step(self, precursors_df: pd.DataFrame, fragments_df: pd.DataFrame):
         """
         pass
 
+
+class AutomaticOptimizer(BaseOptimizer):
+    def __init__(
+        self,
+        initial_parameter: float,
+        workflow,
```

Comment:
maybe add the Args from BaseOptimizer explicitly so we know what they are and they become part of type hints, docstrings.